xbmc/addons/addoninfo/AddonInfo.h

```diff
@@ -138,7 +138,9 @@ class CAddonInfo
    * @param[in] mainOnly to check only in first defined main addon inside addon.xml
    * @return true in case the wanted type is supported, false if not
    */
-  bool HasType(TYPE type, bool mainOnly = false) const;
+  bool HasType(TYPE type, bool mainOnly) const;
+
+  bool HasType(TYPE type) const { return HasType(type, false); }
```

Comment:
Think to use only as one function, like there is the same in code write on other places as with `bool mainOnly = false` before. Only then another call from process work inside (where to new stack works).

To have, think to have more clean can it be then as:
```cpp
enum class TYPE_ASK_WAY
{
  Complete = 0,
  MainOnly = 1,
};

bool HasType(TYPE type, TYPE_ASK_WAY way) const;
```

So that callers must set this and there visible what wanted.